routemaster/services/deliver.rb

```diff
@@ -36,14 +45,24 @@ def call
         error = nil
         start_at = Routemaster.now
         begin
-        # send data
+          _maybe_throttle!
+          @subscriber.attempting_delivery(start_at)
+
+          # send data
           response = _conn.post do |post|
             post.headers['Content-Type'] = 'application/json'
             post.body = Oj.dump(_data, mode: :compat)
           end
-          error = CantDeliver.new("HTTP #{response.status}") unless response.success?
+
+          unless response.success?
+            @subscriber.change_health_by(-2)
+            error = CantDeliver.new("HTTP #{response.status}")
+          end
+        rescue EarlyThrottle => e
+          error = _wrap_error(e)
         rescue Faraday::Error::ClientError => e
-          error = CantDeliver.new("#{e.class.name}: #{e.message}")
+          @subscriber.change_health_by(-2)
```

Comment:
Similarly here: the throttle updates the batch health status in batch mode (by calling `batch.fail`, which has a side effect) — but updating subscriber health is done in the caller?

How about:

```diff
rescue Faraday::Error::ClientError => e
-   @subscriber.change_health_by(-2)
+   _throttle.fail # calls @subscriber.change_health_by or batch.fail depending on mode
```